### PR TITLE
[Merged by Bors] - Rename fixed timestep state and add a test

### LINKED ISSUE #3234: Rename `bevy::core::State`
## How can Bevy's documentation be improved?

Those two types are not the same:
- https://docs.rs/bevy/latest/bevy/core/struct.State.html
- https://docs.rs/bevy/latest/bevy/prelude/struct.State.html

One of them need to be renamed, probably the one from `core`

### PR DESCRIPTION
**Pull request by NiklasEi**
# Objective

fixes #3234

## Solution

- rename `bevy::core::State` to `LocalFixedTimestepState`
- add a test for FixedTimestep since I am already there

### CODE REVIEW

File: crates/bevy_core/src/time/fixed_timestep.rs [hunk 1]
```diff
@@ -194,3 +195,79 @@ impl System for FixedTimestep {
...
+        assert_eq!(0usize, *world.get_resource::<usize>().unwrap());
+        assert_eq!(4., get_accumulator_deciseconds(&world));
+
+        // finish the first step with 0.1s above the step length
+        set_time(&mut world, instance, 0.6);
+        schedule.run(&mut world);
+        assert_eq!(1usize, *world.get_resource::<usize>().unwrap());
+        assert_eq!(1., get_accumulator_deciseconds(&world));
+
+        // runns multiple times if the delta is multiple step lengths
```
Comment (alice-i-cecile) [inline crates/bevy_core/src/time/fixed_timestep.rs:245]:
```suggestion
        // runs multiple times if the delta is multiple step lengths
```

File: crates/bevy_core/src/time/fixed_timestep.rs [hunk 2]
```diff
@@ -194,3 +195,79 @@ impl System for FixedTimestep {
...
+
+    #[test]
+    fn test() {
+        let mut world = World::default();
+        let mut time = Time::default();
+        let instance = Instant::now();
+        time.update_with_instant(instance);
+        world.insert_resource(time);
+        world.insert_resource(FixedTimesteps::default());
+        world.insert_resource(0usize);
```
Comment (alice-i-cecile) [inline crates/bevy_core/src/time/fixed_timestep.rs:217]: I would probably prefer a `Count` newtype for clarity here, but that's just a nit.

**Review by alice-i-cecile — APPROVED** [review]

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_core/src/time/fixed_timestep.rs [hunk 3]
```diff
@@ -194,3 +195,80 @@ impl System for FixedTimestep {
...
+        schedule.run(&mut world);
+        assert_eq!(3, *world.get_resource::<Count>().unwrap());
+        assert_eq!(2., get_accumulator_deciseconds(&world));
+    }
+
+    fn fixed_update(mut count: ResMut<Count>) {
+        *count += 1;
+    }
+
+    fn set_time(world: &mut World, instance: Instant, seconds: f32) {
```
Comment (mockersf) [inline crates/bevy_core/src/time/fixed_timestep.rs:257]: small nit, but this function doesn't set time, it advances time

**Review by mockersf — APPROVED** [review]

**Review by DJMcNab — APPROVED** [review]
I haven't checked the test code, but the renaming looks good.

I want this to land, because I want to refactor `FixedTimestep`.

File: crates/bevy_core/src/time/fixed_timestep.rs (hunk 3 above)
Comment (NiklasEi) [reply crates/bevy_core/src/time/fixed_timestep.rs:257]: It updates Time with an `Instant`, not a `Duration`. That's why it seems more like setting, not advancing to me.
But I got no problem renaming it.

Comment (alice-i-cecile) [reply crates/bevy_core/src/time/fixed_timestep.rs:257]: Yeah, I think I agree with `advance_time` here: it's more clear in the context of this test.

**Pull request closed without merging**